libs/packages/layouts/src/lib/feature/search-list-layout/search-list-layout.component.ts

```diff
@@ -1,63 +1,75 @@
-import { BehaviorSubject, Subscription } from "rxjs";
-import { SearchListInterface, SearchListConfiguration } from './model/search-list-layout.model';
-import { SDSFormlyUpdateComunicationService } from '@gsa-sam/sam-formly';
+import {
+  Component,
+  Input,
+  ContentChild,
+  TemplateRef,
+  Optional,
+  OnChanges,
+  SimpleChanges,
+  OnInit,
+  ChangeDetectorRef,
+  HostListener,
+} from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
+import * as qs from 'qs';
+import {
+  SearchListInterface,
+  SearchListConfiguration,
+  ResultsModel,
+} from './model/search-list-layout.model';
+import {
+  SDSFormlyUpdateComunicationService,
+  SDSFormlyUpdateModelService,
+} from '@gsa-sam/sam-formly';
+import { Router, ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'search-list-layout',
   templateUrl: './search-list-layout.component.html',
-  styleUrls: ['./search-list-layout.component.scss']
+  styleUrls: ['./search-list-layout.component.scss'],
 })
-export class SearchListLayoutComponent implements OnInit, OnDestroy {
-
+export class SearchListLayoutComponent implements OnChanges, OnInit {
   /**
-  * Child Template to be used to display the data for each item in the list of items
-  */
+   * Child Template to be used to display the data for each item in the list of items
+   */
   @ContentChild('resultContent') resultContentTemplate: TemplateRef<any>;
 
-  constructor(@Optional() private formlyUpdateComunicationService: SDSFormlyUpdateComunicationService) { }
-  totalItems = 0;
+  constructor(
+    private cdr: ChangeDetectorRef,
+    private router: Router,
+    private route: ActivatedRoute,
+    @Optional()
+    private formlyUpdateComunicationService: SDSFormlyUpdateComunicationService,
+    private filterUpdateModelService: SDSFormlyUpdateModelService
+  ) {}
+
   /**
    * Input service to be called when items change
    */
-  @Input()
-  service: SearchListInterface;
+  @Input() service: SearchListInterface;
 
- /**
+  /**
    * Allow to insert a customized template for no results to use
    */
   @Input() customResultsTemplate: TemplateRef<any>;
 
+  @Input() isHistoryEnabled = true;
   /**
    * configuration
    */
-  @Input()
-  configuration: SearchListConfiguration;
+  @Input() configuration: SearchListConfiguration;
 
   /**
    * Filter information
    */
   private filterData: any;
 
+  /**
+   * Total number of items
+   */
+  totalItems: number;
   private formlySubscription: Subscription;
 
-  ngOnInit() {
-    this.page.pageSize = this.configuration.pageSize;
-    this.sortField = this.configuration.defaultSortValue;
-    this.paginationChange.subscribe(
-      () => {
-        this.updateContent();
-      }
-    );
-    if (this.formlyUpdateComunicationService) {
-      this.formlySubscription = this.formlyUpdateComunicationService.filterUpdate.subscribe(
-        (filter) => {
```

Comment:
Could you also update the onInit method to have this assignment of formlySubscription. Here, since the subscription comes from a service, it's not auto cleaned up when component is destroyed, so we need to manually unsubscribe during ngOnDestroy